### ISSUE TITLE
Issue with Windows Mixed Reality Teleportation when using nuget packages

### ISSUE THREAD

**Issue by pefritzii** [microsoft/MixedRealityToolkit-Unity#6370]
## Describe the bug
Maybe I missed something in the tutorials and instructions, but I have set up a new project in Unity using MRTK v2.  Everything I have read shows that once imported and set up I should be able to see the gizmos and once I have a collider on my floor I should be able to teleport.  I do not see this at all with Windows Mixed Reality.  No gizmo, no teleportation graphics, and no movement.  I then reset the scene using the HTC Vive and everything worked as it should.  I saw the gizmos and I was able to teleport.  For the WMR I changed the build settings to UWP and set the player settings for Windows Mixed Reality.  When this did not work I went looking for a tutorial in YouTube and followed it (https://www.youtube.com/watch?v=-ODnfcv5Rzg).  Still no luck.  I've tried this on two different computers and two different brands of WMR systems (Lenovo and HP).  
 
A clear and concise description of what the bug is.  See above.

## Your Setup (please complete the following information)

- Unity Version 2019.2.8f1
- MRTK Version v2.0

## Target Platform 
- WMR immersive (HP and Lenovo)

**Comment by Troy-Ferrell** [comment]
@pefritzii what example scene are you testing and/or what MRTK profile do you have selected? Can you confirm the Teleport system is enabled in the MRTK configuration profile you have in scene?

**Comment by pefritzii** [comment]
Hi Troy,

Thanks for getting back to me.  I am able to get this scene and others working with the HTC Vive and the MRTK v2 but not any of the Windows Mixed Reality systems I have tried.  I am including several screen shots if they help.  I am a high school 3D animation teacher.  I use Maya, Blender, Unity, and Unreal Engine in my classes.  We build a city early on in class and use it for a back drop and a prop.  I also use it to teach them about VR.  I am not a high level game programmer, but I know enough usually to solve basic problems.  The buildings in this scene were used last year and I had everything working using the previous MRTK.  Since we upgraded again this year to Unity 2019.2.8f1 and to the MRTK v2 I decided to make sure I knew what I was doing before I got in front of 100 students. I just cannot figure out why the the Windows Mixed Reality is not working.  The Teleport system is checked and it is working with the Vive.  I was thinking maybe it was the old scene causing conflicts so I set up a new scene in Unity and imported the buildings in to test that theory.  No luck there either.  I'm sure I am just missing something simple.  Thanks again! 


Here are the pictures:


 


On October 24, 2019 at 12:00 PM, Troy Ferrell <notifications@github.com> wrote:



@pefritzii what example scene are you testing and/or what MRTK profile do you have selected? Can you confirm the Teleport system is enabled in the MRTK configuration profile you have in scene?

**Comment by Troy-Ferrell** [comment]
@pefritzii Unfortunately the pictures did not come through. You may have to post directly on github.com? 

Can you try matching the following? I was able to get the teleport system working under these conditions. Also check any windows updates etc.

Unity 2019.2.10
MRTK v2.1
Windows 10 1903 - 18362.385
Create empty scene
Add MRTK toolkit (Add to scene & configure)
Select "DefualtMRTKConfigurationProfile" option 

![image](https://user-images.githubusercontent.com/25975362/67590895-f0066880-f710-11e9-8233-c2a9270a0764.png)

Add simple plane to environment
Click Play in editor
Use Samsung Odyssye VR controller to teleport

![teleport](https://user-images.githubusercontent.com/25975362/67591011-3b207b80-f711-11e9-9ec1-10f74bc3acf0.gif)

**Comment by pefritzii** [comment]
Troy,

I mimicked what you have here as much as possible.  I am limited by the set up here at work.  So, I have Unity 2019.2.8 and Windows  1803 build 17134.1069.  I did use NuGet to install the MRTK which is v2.1.  I am still not able to get Windows MR to work.  Again, using this configuration I was able to get the Vive to work.  I have also set this up at home with the current build of Windows 10.  I have not updated Unity to 2019.2.10.  I will try that later today.  


Thanks again.



Paul


On October 25, 2019 at 12:22 PM, Troy Ferrell <notifications@github.com> wrote:



@pefritzii Unfortunately the pictures did not come through. You may have to post directly on github.com?

Can you try matching the following? I was able to get the teleport system working under these conditions. Also check any windows updates etc.

Unity 2019.2.10
MRTK v2.1
Windows 10 1903 - 18362.385
Create empty scene
Add MRTK toolkit (Add to scene & configure)
Select "DefualtMRTKConfigurationProfile" option




Add simple plane to environment
Click Play in editor
Use Samsung Odyssye VR controller to teleport

**Comment by Troy-Ferrell** [comment]
@keveleigh as FYI in case he has any good insights. 

Can you also maybe elaborate on what isn't working in more detail?

Do other features work for WMR? Do the WMR controllers appear in play mode? Are there any warnings, notices or errors logged in Unity console when you run the experience? Can you select items in the HandsInteractionExample scene with controllers? Does the boundary system visualization appear( check if the system is enabled in your MRTK profile as well)?

**Comment by pefritzii** [comment]
Troy,

Here is an update from my earlier response.  I set up yet another scene but this time did not use the NuGet installer and imported the MRTK directly.  Once I did this everything worked as it should.  I am going to go back and try it in my pre-built scene. 



Paul


On October 25, 2019 at 12:22 PM, Troy Ferrell <notifications@github.com> wrote:



@pefritzii Unfortunately the pictures did not come through. You may have to post directly on github.com?

Can you try matching the following? I was able to get the teleport system working under these conditions. Also check any windows updates etc.

Unity 2019.2.10
MRTK v2.1
Windows 10 1903 - 18362.385
Create empty scene
Add MRTK toolkit (Add to scene & configure)
Select "DefualtMRTKConfigurationProfile" option




Add simple plane to environment
Click Play in editor
Use Samsung Odyssye VR controller to teleport

**Comment by Troy-Ferrell** [comment]
> I set up yet another scene but this time did not use the NuGet installer and imported the MRTK directly.  Once I did this everything worked as it should.  I am going to go back and try it in my pre-built scene. 

Fantastic! Glad to hear we have narrowed it down and sounds like at least at this moment you are unblocked. I know this must have been very frustrating. 

@julenka, possible similar issue as your #6391 

@andreiborodin as FYI at likely nuget issue

**Comment by pefritzii** [comment]
What isn't working is the teleportation.  I have no errors or warnings in the console.  The gizmos do not show up and no teleportation happens.  When I walk up to the edge of the boundaries the controllers are detected by the boundaries.  


On October 28, 2019 at 2:54 PM, Troy Ferrell <notifications@github.com> wrote:



@keveleigh as FYI in case he has any good insights.

Can you also maybe elaborate on what isn't working in more detail?

Do other features work for WMR? Do the WMR controllers appear in play mode? Are there any warnings, notices or errors logged in Unity console when you run the experience? Can you select items in the HandsInteractionExample scene with controllers? Does the boundary system visualization appear( check if the system is enabled in your MRTK profile as well)?

**Comment by pefritzii** [comment]
Thank you for the help.  I usually work through these issues on my own, but this one stumped me because of the simplicity of the set up.  I'll have to dig into the NuGet to see if there is something that I missed when it goes out and gets the WMRT files. 


On October 28, 2019 at 3:18 PM, Troy Ferrell <notifications@github.com> wrote:



I set up yet another scene but this time did not use the NuGet installer and imported the MRTK directly. Once I did this everything worked as it should. I am going to go back and try it in my pre-built scene.

Fantastic! Glad to hear we have narrowed it down and sounds like at least at this moment you are unblocked. I know this must have been very frustrating.

@julenka, possible similar issue as your #6391


@andreiborodin as FYI at likely nuget issue

**Comment by julenka** [comment]
I think this is the same as #6391, I also see no response at all to WMR controlelrs using NuGet. Using MRTK unitypackages works. duplicate of #6391 I'll close this one so we can track in #6391 which has the root cause (WMR controllers don't work at all with NuGet)

**Comment by pefritzii** [comment]
Yes, I believe that is correct.  However, the boundaries do seem to detect the controllers even if they are not visible and do not teleport.
